beta/src/pages/apis/usestate.md

```diff
@@ -87,12 +87,16 @@ function handleClick() {
 
 `set` functions do not have a return value.
 
-#### Caveats {/*caveats*/}
+#### Caveats {/*setstate-caveats*/}
 
 * The `set` function only updates the state variable for the next render. If you read the state variable after calling the `set` function, you will still get the value that was on the screen before your call.
 
 * If the new value you provide is identical to the current `state`, as determined by an [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison, React will bail out of re-rendering the component and its children. However, in some cases React may still need to call your component before discarding the result.
 
+* React batches state updates: it does not update the screen until right after all the event handlers have called their `set` functions. This prevents multiple re-renders during a single event. In the rare case that you need to force React to update the screen earlier so that you can access the DOM, you can use [`flushSync`](/apis/flushsync).
+  - *In React 17 and earlier,* batching is only enabled for the `set` function calls during React-managed event handlers, like the ones you put in your JSX. React updates the screen at the end of the browser event. Updates from two distinct events never get batched together. 
+  - *In React 18 and later,* batching is enabled for all `set` function calls. This includes both React event listeners and listeners you've added manually with `addEventListener`. React updates the screen in the next [microtask](https://developer.mozilla.org/en-US/docs/Web/API/HTML_DOM_API/Microtask_guide). Updates from two distinct *intentional* events like clicks or typing never get batched together. 
+
 * Calling the `set` function *during rendering* is only allowed from within the currently rendering component. React will discard its output and immediately attempt to render it again with the new state. React will do this in a loop until you either stop setting state, or do it too many times. This pattern is called "derived state". It is uncommon, but you can use it to store information from previous renders. [See an example below.](#storing-information-from-previous-renders)
 
 * In Strict Mode, React will call your updater function twice in order to help you find accidental impurities. This is development-only behavior and does not affect production. If your updater function is pure (as it should be), this should not affect the logic of your component. The result from one of the calls will be ignored.
```

Comment:
Do you think we could link the "accidental impurities" to the learn section on pure functions? I just worry that some readers may not know about what constitutes impurity.